Libmesh submodule update

roystgnr opened the pull request:
This shouldn't be merged yet, since we don't want to make @milljm 's job in #26848 even harder; plus there's another libMesh PR that I'd be happy to see snuck in under the wire if we really take too long ... but I'm anxious like to see what downstream CI makes of the changes that are here so far; it's entirely possible that we have users who are reading QuadShell4 meshes (because that's a Cubit default) and turning them second-order, and any golded results based on that will break when "full second-order" becomes *actually*-full second-order - see #27987 for the fix to such a failure in the Navier-Stokes module.

Summary of changes:

- Support for `QuadShell9` elements, and fixes for refinement of
  `QuadShell8` elements.  This is a behavior-changing update:
  `QuadShell4` elements, when elevated to second order, now become
  `QuadShell9` (with a full biquadratic Lagrange space) rather than
  `QuadShell8` (with a "Serendipity element" space).
- Fixes for libMesh `make dist` and `make distcheck` with new
  submodules and new contributed packages.
- Support and unit tests for non-square EigenSparseMatrix matrices
- Support and unit tests for reading sparse matrices from
  gzipped-Matlab format files.
- Utility for converting sparse matrix files from PETSc binary formats
  to Matlab format.
- Improved speed for reading sparse matrices.
- Refactoring of some solution transfer code; exposure of
  `MeshfreeInterpolationFunction` class to users.
- Support and unit tests in TIMPI for `set_union` of `multiset` and
  `multimap` containers, including `unordered_` versions.
- Refactored Nedelec finite element code for simplification and
  generality.
- Fixes and testing for more IsoGeometricAnalysis use cases on
  distributed meshes.
- Additional fparser headers installed, to enable user subclassing
- Assorted bug fixes: configure now supports preexisting `-Werror` in
  compiler flags (with tested compilers), more standard treatment of
  PETSc error codes.
- PetscVector subvectors can now be created by supplying only local
  subvector indices.
- Incoming vectors to `local_variable_indices()` are now cleared
  before filling.
- PerfLog profiling of matrix reads and mesh constraint
  initialization.
- Refactoring and optional debugging features for mesh constraint
  code.
- Added `RBConstruction::is_serial_training_type()` and EIM updates to
  reduced basis code.
- Added a DofMap API for users to request full matrix sparsity pattern
  retention even with linear algebra packages that do not need it
  themselves.

roystgnr committed:
Newsletter entries for libMesh submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2303531/) on c40ae34 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 1c5369aa05eaae8ebeb7930845fce3b1d59cbb37
+Subproject commit 9820ed83058a17cd1e33e0f49852f699c3430792
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2303531/) on c40ae34 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2303531/) on c40ae34 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr commented:
👔

On Fri, Jun 28, 2024, 10:34 AM MOOSE Build ***@***.***> wrote:

> Job Precheck <https://civet.inl.gov/job/2303531/> on c40ae34
> <https://github.com/idaholab/moose/commit/c40ae34721aa831a9b2635547b65f7ec51b47349>
> wanted to post the following:
>
> The following file(s) are changed:
>
> conda/libmesh/meta.yaml
>
> The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops
> <https://github.com/orgs/idaholab/teams/moose-dev-ops>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/idaholab/moose/pull/28014#issuecomment-2197189899>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AASB64TN2V6YDOTPVQJ4NE3ZJV7BFAVCNFSM6AAAAABKCDKQUWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOJXGE4DSOBZHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2303527/) on c40ae34 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/28014/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2303571/) on c40ae34 wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/28014/coverage/xfem)

</details>



This comment will be updated on new commits.

roystgnr commented:
Rebased and hashes redone, just to see if CI gives us any new headaches, but we'll still also want to upgrade to a slightly newer libMesh hash shortly.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2330975/) on 46953a7 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2333626/) on 1c95e4f wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Valgrind contact](https://civet.inl.gov/job/2333691/) on 1c95e4f : invalidated by @roystgnr

Kicking valgrind timeouts

roystgnr committed:
More libMesh update newsletter entries

roystgnr committed:
Update libMesh submodule

Refs #0

Summary of changes:

- Support for `QuadShell9` elements, and fixes for refinement of
  `QuadShell8` elements.  This is a behavior-changing update:
  `QuadShell4` elements, when elevated to second order, now become
  `QuadShell9` (with a full biquadratic Lagrange space) rather than
  `QuadShell8` (with a "Serendipity element" space).
- Fixes for libMesh `make dist` and `make distcheck` with new
  submodules and new contributed packages.
- Support and unit tests for non-square EigenSparseMatrix matrices
- Support and unit tests for reading sparse matrices from
  gzipped-Matlab format files.
- Utility for converting sparse matrix files from PETSc binary formats
  to Matlab format.
- Improved speed for reading sparse matrices.
- Refactoring of some solution transfer code; exposure of
  `MeshfreeInterpolationFunction` class to users.
- Support and unit tests in TIMPI for `set_union` of `multiset` and
  `multimap` containers, including `unordered_` versions.
- Refactored Nedelec finite element code for simplification and
  generality.
- Fixes and testing for more IsoGeometricAnalysis use cases on
  distributed meshes.
- More test coverage: `ExodusII_IO` with shell elements, PETSc
  DMlibMesh interface
- New `Elem::is_internal()` API for local node indices
- Additional fparser headers installed, to enable user subclassing
- Assorted bug fixes: configure now supports preexisting `-Werror` in
  compiler flags (with tested compilers), more standard treatment of
  PETSc error codes, `--node-major-dofs` now works with FE types that
  have element DoFs too.
- Added `LIBMESH_PETSC_SUCCESS` shim
- PetscVector subvectors can now be created by supplying only local
  subvector indices.
- Slight change to `restore_subvector()` arguments
- Incoming vectors to `local_variable_indices()` are now cleared
  before filling.
- PerfLog profiling of matrix reads and mesh constraint
  initialization.
- Refactoring and optional debugging features for mesh constraint
  code.
- Added `RBConstruction::is_serial_training_type()` and EIM updates to
  reduced basis code.
- Added a DofMap API for users to request full matrix sparsity pattern
  retention even with linear algebra packages that do not need it
  themselves.
- `System::is_initialized()` is now `const`

roystgnr committed:
Update versioner_hashes.yaml

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 1c5369aa05eaae8ebeb7930845fce3b1d59cbb37
+Subproject commit 81a423f6eaa813c5d6d3a92eb2a24cc7cc2b8e34
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2337214/) on 6360ad7 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

lindsayad commented on modules/doc/content/newsletter/2024/2024_07.md:
```diff
@@ -9,6 +9,54 @@ for a complete description of all MOOSE changes.
+  retention even with linear algebra packages that do not need it
+  themselves.
+- `System::is_initialized()` is now `const`
+
```
```suggestion
- Added `LibmeshPetscCall` macro that can wrap any PETSc function call without need for `LIBMESH_CHKERR(ierr)`

```

lindsayad reviewed (commented):
This will be great to get in. A lot of nice changes/additions. This is needed for the PETSc update

roystgnr committed:
Update modules/doc/content/newsletter/2024/2024_07.md

Co-authored-by: Alex Lindsay <alexlindsay239@gmail.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2339098/) on c39ebbc wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour commented:
Feel free to merge @roystgnr if I'm not around to do it. Just check next and make sure it's passing first, if so.

lindsayad commented:
Noooo I should have checked this more closely. This doesn't have libmesh/libmesh#3905

loganharbour commented:
> Noooo I should have checked this more closely. This doesn't have [libMesh/libmesh#3905](https://github.com/libMesh/libmesh/pull/3905)

Put another up. Our build infrastructure for these environments has gotten a lot cleaner and I'm on board with weekly updates if there's a need.

roystgnr commented:
Going to squeeze https://github.com/libMesh/libmesh/pull/3895 and hopefully an upcoming PR of my own in before the next submodule update, but that should still be this week.
